Source/common/SNTLogging.h

```diff
@@ -1,59 +1,72 @@
 /// Copyright 2015 Google Inc. All rights reserved.
+/// Copyright 2025 North Pole Security, Inc.
 ///
 /// Licensed under the Apache License, Version 2.0 (the "License");
 /// you may not use this file except in compliance with the License.
 /// You may obtain a copy of the License at
 ///
-///    http://www.apache.org/licenses/LICENSE-2.0
-///
-///    Unless required by applicable law or agreed to in writing, software
-///    distributed under the License is distributed on an "AS IS" BASIS,
-///    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-///    See the License for the specific language governing permissions and
-///    limitations under the License.
-
-///
-/// Logging definitions
+///     https://www.apache.org/licenses/LICENSE-2.0
 ///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
 
 #ifndef SANTA__COMMON__LOGGING_H
 #define SANTA__COMMON__LOGGING_H
 
-#ifdef __cplusplus
-extern "C" {
-#endif
-
 #import <Foundation/Foundation.h>
+#include <os/log.h>
+#include <sys/cdefs.h>
 
-typedef enum : NSUInteger {
-  LOG_LEVEL_ERROR,
-  LOG_LEVEL_WARN,
-  LOG_LEVEL_INFO,
-  LOG_LEVEL_DEBUG
-} LogLevel;
+__BEGIN_DECLS
 
-///
-///  Logging function.
-///  @param level one of the levels defined above
-///  @param destination a FILE, generally stdout/stderr. If the file is closed, the log
-///      will instead be sent to syslog.
-///  @param format the printf style format string
-///  @param ... the arguments to format.
-///
-void logMessage(LogLevel level, FILE *destination, NSString *format, ...)
-    __attribute__((format(__NSString__, 3, 4)));
+#define SNT_LOG_WITH_TYPE(type, fmt, ...)              \
+  os_log_with_type(OS_LOG_DEFAULT, type, "%{public}s", \
+                   [[NSString stringWithFormat:fmt, ##__VA_ARGS__] UTF8String])
+
+#define SNT_PRINT_LOG(file, fmt, ...) \
+  fprintf(file, "%s\n", [[NSString stringWithFormat:fmt, ##__VA_ARGS__] UTF8String]);
+
+#define LOGD(logFormat, ...) SNT_LOG_WITH_TYPE(OS_LOG_TYPE_DEBUG, logFormat, ##__VA_ARGS__)
+#define LOGI(logFormat, ...) SNT_LOG_WITH_TYPE(OS_LOG_TYPE_INFO, logFormat, ##__VA_ARGS__)
+#define LOG(logFormat, ...) SNT_LOG_WITH_TYPE(OS_LOG_TYPE_DEFAULT, logFormat, ##__VA_ARGS__)
```

Comment:
I don't like there being a 'bare' LOG without a level. It won't be clear in context whether that's higher or lower than INFO or WARN, for instance, and the 4 levels we have have been plenty. If this exists just to use the OS_LOG_TYPE_DEFAULT level, why not map LOGW to default?

Reply:
Was added because "warn" doesn't map cleanly in the os_log world. While originally I mapped LOGW to OS_LOG_TYPE_DEFAULT, this looked odd when adding TEE_LOGW which would write to stderr. However, thinking on it more, I think this is fine. "Warn" messages are not errors, so no need to use the ERROR log type. And for the TEE variant, we can still write to stderr so warn-level logs don't interfere with stdout.